[appleraincoat] iP

appleraincoat opened the pull request:
# Ypxmm
> If you fail to plan, you plan to fail.

Ypxmm is your go-to, true-blue Singaporean task manager bot. It is:

- text-based
- user-friendly
- very fast to use

## How to use

1. Download it from [here](https://github.com/appleraincoat/ip/releases/tag/A-Release)
2. Open the folder with CLI
3. Run ```java -jar Ypxmm.jar```
4. Add your tasks

And you are done! 😄 

## Features
- [x] Add 3 types of tasks, Todos, Deadlines and Events
- [X] List, Mark & Unmark and Delete tasks
- [ ] Assign priorities to tasks (Coming soon)
- [ ] Edit tasks (Coming soon)

## Implementation

This program was written in Java. The main class is as follows.

```
public class Launcher {
    public static void main(String[] args) {
        Application.launch(Main.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

appleraincoat committed:
Added chatbot Ypxmm, with only simple greetings before exiting

appleraincoat committed:
Added the ability to echo messages, exiting on input "bye".

appleraincoat committed:
Removed ability to echo, keeps a lsit of inputs now and prints it upon request.

appleraincoat committed:
Abstracted a Task class.

appleraincoat committed:
Add and extend 3 types of Tasks, Deadline, Event and Todo. Add static methods into main class to abstract out further.

appleraincoat committed:
Add some simple test cases for semi-automated testing.

appleraincoat committed:
Add handling of invalid input for mark and unmark methods.

appleraincoat committed:
Add YpxmmException class. Able to handle invalid inputs.

appleraincoat committed:
Add more exception handling features.

appleraincoat committed:
Add test cases for error handling.

appleraincoat committed:
Add delete function.

appleraincoat committed:
Add new test cases.

appleraincoat committed:
Add Enum class for commands

appleraincoat committed:
Add enums, and change implementation to use them in a switch statement. Add 1 more catch for illegal argument exception. Add more test cases.

appleraincoat committed:
Change import statements in Ypxmm.

appleraincoat committed:
Add ability to save tasks to a file in hard disk, and load from it when initialise.

appleraincoat committed:
Add trimming when initialising tasks.

appleraincoat committed:
Merge branch 'branch-Level-7'

appleraincoat committed:
Change byWhen variable in Deadline class to store in LocalDateTime format.

appleraincoat committed:
Change Deadline and Event class to store timing with LocalDateTime type, adjust formatting to be consistent.

appleraincoat committed:
Modify getcommands command to suit the change in format

appleraincoat committed:
Add a catch for DateTimeParseException if the user inputs in the wrong format

appleraincoat committed:
Merge branch 'branch-Level-8'

appleraincoat committed:
Refactor code. Rename EnumCommands to Command, add Parser, Storage, Ui, TaskList classes.

appleraincoat committed:
Small bug fixes

appleraincoat committed:
Add DateTImeParser class that parses format of time in Deadline and Event classes. Add other small modifications to UI output.

appleraincoat committed:
Merge branch 'A-MoreOOP'

appleraincoat committed:
Package all classes into their respective packages

appleraincoat committed:
Merge branch 'A-Packages'

appleraincoat committed:
Merge remote-tracking branch 'origin/add-gradle-support'

appleraincoat committed:
Fix small error in build.gradle.

appleraincoat committed:
Add some test classes for Command and DateTimeParser classes and unpackage main class Ypxmm. Modify gradle build, set the name and mainclass to Ypxmm.

appleraincoat committed:
Add tests for Parser class. Tests for Command and DateTimeParser classes incomplete.

appleraincoat committed:
Remove CommandTest for now, add DateTimeParserTest class and DateTimeParserStub to help in testing.

appleraincoat committed:
Merge branch 'A-JUnit'

appleraincoat committed:
Configure to create jar file.

appleraincoat committed:
Merge branch 'A-Jar'

appleraincoat committed:
Add a line in getcommands to show the delete command.

appleraincoat committed:
Add Find command, which finds tasks with that keyword in their name.

appleraincoat committed:
Add JavaDocs for DateTimeParserClass.

appleraincoat committed:
Adjust coding standard for DateTimeParser class.

appleraincoat committed:
Fix some coding standard violations.

appleraincoat committed:
Add JavaDocs for all classes and methods.

appleraincoat committed:
Merge branch 'A-JavaDoc'

appleraincoat committed:
Merge branch 'A-CodingStandard'

# Conflicts:
#	src/main/java/Parsing/DateTimeParser.java
#	src/main/java/Parsing/ParserStub.java
#	src/main/java/Ypxmm.java

appleraincoat committed:
Merge branch 'Level-9'

# Conflicts:
#	src/main/java/Commands/Command.java
#	src/main/java/Parsing/Parser.java
#	src/main/java/Parsing/ParserStub.java
#	src/main/java/Tasks/Task.java
#	src/main/java/Utilities/Storage.java
#	src/main/java/Utilities/Ui.java
#	src/test/java/Parsing/ParserTest.java

Merge all files. Add JavaDocs for Level 9 methods which do not have yet.

tejas-1405 commented on src/main/java/Tasks/Task.java:
```diff
@@ -0,0 +1,78 @@
+    /**
+     * Sets the task as completed.
+     */
+    public void setCompleted() {
```
the setCompleted() and unmarkTask() could be standardised and have similar names since their functions are similar

tejas-1405 commented on src/main/java/Tasks/Task.java:
```diff
@@ -0,0 +1,78 @@
+public class Task {
+
+    protected String name;
+    protected boolean completed;
```
Variable completed could be changed to isCompleted

tejas-1405 commented on src/main/java/Utilities/TaskList.java:
```diff
@@ -0,0 +1,82 @@
+     * @param task the task to add
+     * @throws YpxmmException if an error occurs while adding the task
+     */
+    public void addTask(ArrayList<String> input, Task task) throws YpxmmException {
```
Good use of JavaDocs

tejas-1405 commented on src/main/java/Utilities/Ui.java:
```diff
@@ -0,0 +1,142 @@
+     * Displays available commands.
+     */
+    public void getCommands() {
+        System.out.println("Take note ah, enter all time based commands are in <dd-mm-yyyy HHmm> format\n" +
```
Great line formatting!

tejas-1405 reviewed (commented):
Overall looks good.

logical-1985516 commented on src/main/java/Parsing/Parser.java:
```diff
@@ -0,0 +1,96 @@
+            } catch (IndexOutOfBoundsException e) {
+                throw new YpxmmException("What u want me to find??");
+            }
+        case "mark":
+        case "unmark":
```
Should there be // Fallthrough comments here?

logical-1985516 commented on src/main/java/Parsing/Parser.java:
```diff
@@ -0,0 +1,96 @@
+        ArrayList<String> parsedResult = new ArrayList<>();
+        String[] splitInput = input.split(" ");
+        String command = splitInput[0];
+        switch (command) {
+        case "bye":
+        case "list":
+        case "getcommands":
```
I like that the case statements are not indented by 4 spaces. Good job!

logical-1985516 commented on src/main/java/Commands/Command.java:
```diff
@@ -0,0 +1,278 @@
+     * Command to list all tasks.
+     */
+    LIST {
+        /**
+         * Executes the command to list all tasks.
+         *
+         * @param tasklist the list of tasks
+         * @param ui the user interface
+         * @param storage the storage utility
+         * @param parsed the parsed command arguments
+         */
```
I like that the Javadocs are in the correct form (especially the singular form verb in the opening sentence). Well done!

logical-1985516 reviewed (commented):
Coding standards is mostly adhered to. Good job!

appleraincoat committed:
Add JavaFX implementation and basic GUI features

appleraincoat committed:
Add some styling changes to the GUI

appleraincoat committed:
Fix some small issues with the truncation when listing tasks. Make picture round, and fix some small issues with some printing message functions

appleraincoat committed:
Merge branch 'Level-10'

appleraincoat committed:
Update README.md

appleraincoat committed:
Update README.md

appleraincoat committed:
Fix some coding standard violations and rename packages

appleraincoat committed:
Add some Javadocs

appleraincoat committed:
Merge remote-tracking branch 'origin/master'

appleraincoat committed:
Merge pull request #1 from appleraincoat/TestPull

Test Pull

appleraincoat committed:
Fix more coding style violations

appleraincoat committed:
Fix all coding style violations

appleraincoat committed:
Merge branch 'A-Checkstyle'

appleraincoat committed:
Add some assertions to some methods

appleraincoat committed:
Add continuous integration

appleraincoat committed:
Merge branch 'A-CI'

appleraincoat committed:
Merge branch 'A-Assertions'

appleraincoat committed:
Parser class: Create methods for parsing commands

Using only one method for parsing is too long.

Abstracting out these methods improves code quality as it adheres to
the SLAP principle.

Let's create more methods that parses each type of command respectively
for better abstraction in each method.

appleraincoat committed:
Add java docs for the new methods

appleraincoat committed:
Modify java docs and fix some nits

appleraincoat committed:
Merge branch 'A-CodeQuality'

appleraincoat committed:
Modify getcommands command

appleraincoat committed:
Add functionality to remind user of commands they can use as a type
of help

appleraincoat committed:
Merge branch 'C-Help'

appleraincoat committed:
Fix nits with the testing

appleraincoat committed:
Improve the GUI

appleraincoat committed:
Update README.md

appleraincoat committed:
Update README.md

appleraincoat committed:
Merge remote-tracking branch 'origin/master'

appleraincoat committed:
Add ability to attach a priority level to tasks and sort the list
by priority level

appleraincoat committed:
Merge branch 'C-Priority'

appleraincoat committed:
Add help for the new prioritise command

appleraincoat committed:
Fix small check style issues

appleraincoat committed:
Add Ui.png

appleraincoat committed:
Add user guide in the README file

appleraincoat committed:
Modify screenshot names and some spacing issues

appleraincoat committed:
Modify README

appleraincoat committed:
Abstract out methods in ParserStub class to adhere to SLAP principle
for better code quality

appleraincoat committed:
Abstract out method in the Storage class for better code quality

appleraincoat committed:
Merge branch 'A-MoreCodeQuality'

appleraincoat committed:
Fix a small issue where the txt file is created in the root directory
instead
